client/BootstrapClient.java

```diff
@@ -1,16 +1,25 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
 import java.net.*;
+import java.util.Date;
 import java.util.Scanner;
 import java.util.Random;
 import java.io.BufferedReader;
 import java.io.InputStreamReader;
 
 public class BootstrapClient {
 
+    private static final String LOG_FILE = "client_log.txt";
+    private static int hopCounter = 0;
+
     public static void main(String[] args) {
         Scanner scanner = new Scanner(System.in);
-        try {
-            DatagramSocket socket = new DatagramSocket();
-            InetAddress address = InetAddress.getByName("localhost"); // Change to server's address if needed
+        try (DatagramSocket socket = new DatagramSocket();
+             FileWriter fileWriter = new FileWriter(LOG_FILE, true); // Append mode
+             PrintWriter logWriter = new PrintWriter(fileWriter)) {
+
+            InetAddress address = InetAddress.getByName("localhost");
```

Comment:
This should be reverted, as I mentioned in an earlier PR as well, @shafraz17 & myself agreed to have the server's static IP hardcoded in here. However, localhost is totally incorrect. 
please check previous version & add the server's static public IP here.